x/stakeibc/keeper/msg_server_restore_interchain_account.go

```diff
@@ -53,6 +53,13 @@ func (k msgServer) RestoreInterchainAccount(goCtx context.Context, msg *types.Ms
 			return nil, types.ErrHostZoneNotFound.Wrapf("delegation ICA supplied, but no associated host zone")
 		}
 
+		// Since any ICAs along the original channel will never get relayed,
+		// we have to reset the delegation_changes_in_progress field on each validator
```

Comment:
This makes sense if we're restoring the delegation account channel. 

But if we're restoring another channel like the withdrawalAccount, do we want to set these `DelegationChangesInProgress` to 0?

Reply:
I think the delegation channel is the only relevant one cause the only ICAs that change our delegations go through that channel (but please double check me)

Reply:
Sorry, I don't understand. 

Afaiu we have multiple ICA channels to multiple ICAs on the host zone. If any of them close, we need to restore them.

The only way to restore them is via this function.

But we ONLY want to reset the DelegationChangesInProgress to 0 if we're restoring the Delegation ICA channel.
However this fn resets DelegationChangesInProgress to 0 for any channel restoration (E.g. withdrawal channel)

Reply:
this is actually nested under a check that only runs for the delegation channel! [ref]

Easy to miss with the GH diffs haha

Reply:
great call out though - appreciate the scrutiny!